src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -423,7 +557,14 @@ private void BackgroundLogger()
                     try
                     {
                         var maxFileSize = MaxFileSizeInBytes;
-                        using (var currentFile = GetNewStream(maxFileSize))
+                        if (GetNewStream(maxFileSize, out var currentFile) == false)
```

Comment:
I do not like this approach, because logging will stop entirely when we will not be able to get the list of files in the directory. I think it would be safer just to omit the cleanup part. Also let's rename the method, now it does a bit more than the name suggests.